### ISSUE TITLE
[Feat]: [UiAutomator2] expose navigation bar height

### ISSUE THREAD

**Issue by eglitise** [appium/appium#18912]
**Is your feature request related to a problem? Please describe.**
I'm researching ways to improve the accuracy of element highlighters in Appium Inspector, and I noticed that on Android, calculations do not take into account the device navigation bar height. As it turns out, this value is not actually exposed by UiAutomator2.

**Describe the solution you'd like**
Expose the navigation bar height and add it to the session capabilities. From what I can tell, it should be returned by the `/appium/device/system_bars` endpoint, along with the status bar height ([reference](https://github.com/appium/appium/blob/master/packages/base-driver/docs/mjsonwp/protocol-methods.md?plain=1#L136)), but currently only the status bar height is retrieved ([reference](https://github.com/search?q=repo%3Aappium%2Fappium-uiautomator2-driver%20system_bars&type=code)).

**Describe alternatives you've considered**
Hardcoding the navbar height is not a good solution. I have not checked whether it differs depending on device (it probably does), but there would still be at least 2 cases to cover (iOS-style gesture navbar, and 3-button navbar)

**Additional context**
This is just one of the dimensions-related UiA2 issues I've noticed so far, but this feature will greatly help with further investigation.

**Comment by jlipps** [comment]
by navigation bar do you mean the bar at the bottom with back/home buttons?

**Comment by eglitise** [comment]
Yes, exactly that. I assume the endpoint I mentioned does actually provide this value, so it would be very useful to have it included in the session capabilities.

**Comment by jlipps** [comment]
Unfortunately when reading the Uiautomator 2 server source code it looks like only the status bar height is returned, not the navigation bar height. Maybe @mykola-mokhnach knows more about what might be possible here.

**Comment by eglitise** [comment]
I just found [this gist](https://gist.github.com/hamakn/8939eb68a920a6d7a498), if it still works, this could be a possible implementation.

**Comment by jlipps** [comment]
Nice! Seems like that would a pretty straightforward add to the uiauto2 server in that case.

**Comment by jlipps** [comment]
Some of the comments do imply it might not be that reliable across device types

**Comment by mykola-mokhnach** [comment]
@eglitise Have you tried https://github.com/appium/appium-uiautomator2-driver#mobile-getsystembars ?

**Comment by eglitise** [comment]
@mykola-mokhnach thanks, I was unaware of this method. Unfortunately it's not always accurate (potentially for the same reason as in the gist I linked above):
* The height value for both iOS-style gesture navbar and 3-button navbar is the same, and matches that of the 3-button navbar. However, it does correctly change depending on the device orientation
* The visibility property for both bars seems to always be `true`, even when both of them are hidden

I also realised that since these values may change, it would not make sense to include them in session capabilities. Perhaps it is therefore better to instead remove the `viewportRect` and `statBarHeight` properties from capabilities, since they can also change? The replacements would be `driver.getWindowRect` and. `mobile: getSystemBars`, respectively.
XCUITest driver could probably also be adapted by setting `includeDeviceCapsToSessionInfo` to `false` by default.

**Comment by mykola-mokhnach** [comment]
I also don't really the fact these values are included into the session info and would be happy to remove them from there. As far as I remember these were added by @jlipps. So I would prefer to check with Jonathan first about what we'd like to do with them.

As for the values preciseness - we read them from the actual adb output, so it must be Android OS itself providing a fake info.

**Comment by jlipps** [comment]
I don't remember exactly why I added these now, so it is probably safe to remove. But we should be careful to note their removal as a breaking change / major version update in the drivers.
